src/session.cpp

```diff
@@ -246,20 +231,42 @@ session_t::inject(const dispatch_ptr_t& dispatch) {
 
 void
 session_t::revoke(uint64_t channel_id) {
-    channels->erase(channel_id);
+    channels.apply([&](channel_map_t& mapping) {
+        auto it = mapping.find(channel_id);
+
+        // NOTE: Not sure if that can ever happen, but that's why people use asserts, right?
+        BOOST_ASSERT(it != mapping.end());
```

Comment:
We always disable asserts for some reasons. I suggest to drop this check https://github.com/cocaine/cocaine-core/blob/v0.12/include/cocaine/common.hpp#L36